File: leetcode-1-Two_Sum/leetcode-1-Two_Sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        nums_dict: dict = {}

        for n in range(len(nums)):
            nums_dict[nums[n]] = n
        
        for n in range(len(nums)):
            rest: int = (target-nums[n])
            if rest in nums_dict and n!=nums_dict[rest]:
                return [n,nums_dict[rest]]
        return []


if __name__ == "__main__":
    nums: list = [2,7,11,15]
    target: int = 9

    res: list = Solution().twoSum(nums, target)

    print(res)

File: leetcode-1-Two_Sum/leetcode-1-Two_Sum.cs
using System;
using System.Collections.Generic;


public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> vals_inds = new Dictionary<int, int>();
        int[] res = new int[2];
        
        for(int i = 0; i<nums.Length; ++i)
        {
            int diff = (target-nums[i]);
            if(vals_inds.ContainsKey(diff))
            {
                res[0] = i;
                res[1] = vals_inds[diff];
                return res;
            }
            vals_inds[nums[i]] = i;
        }
        return res;
    }
}


public class Program
{
    public static void Main(string[] args)
    {
        int[] nums = {2,7,11,15};
        int target = 9;
        
        Solution solution = new Solution();
        int[] res = solution.TwoSum(nums, target);
        
        Console.Write("[");
        for(int i = 0; i<res.Length; ++i)
        {
            Console.Write(res[i]);
            if(i<res.Length-1)
            {
                Console.Write(", ");
            }
        }
        Console.Write("]");
    }
}